_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -5524,11 +5526,17 @@ mfxStatus ConfigureExecuteParams(
                         if (videoParam.ExtParam[i]->BufferId == MFX_EXTBUFF_VPP_MIRRORING)
                         {
                             mfxExtVPPMirroring *extMirroring = (mfxExtVPPMirroring*) videoParam.ExtParam[i];
-                            if (extMirroring->Type != MFX_MIRRORING_DISABLED && extMirroring->Type != MFX_MIRRORING_HORIZONTAL)
+                            if (extMirroring->Type != MFX_MIRRORING_DISABLED && extMirroring->Type != MFX_MIRRORING_HORIZONTAL
+#if defined(MFX_VA_LINUX)
```

Comment:
I would suggest to avoid macro-if and use runtime level `core->GetVAType() == MFX_HW_VAAPI`.

Reply:
Thanks for the comment, I will update the patch.

Reply:
core is not available in ConfigureExecuteParams(), I updated VideoVPPHW::SyncTaskSubmission() only.